### ISSUE TITLE
Pass keywords to colorbar construction

### ISSUE THREAD

**Issue by b-tierney** [geopandas/geopandas#697]
Hey all,
Amazing tool – thank you so much for putting it together and managing it so well.

I'm sure there is an easy fix to this, but I'm having trouble accessing the legend object from a chloropleth plot. I'd imagine ax.get_legend() should return it, but that doesn't seem to be the case.

My endgoal is simply to put the colormap horizontally beneath my figure – so if there's a very easy way to do that let me know. Thank you so much!

**Comment by mrahim** [comment]
Hi @b-tierney ,

your question looks more related to colorbars/legends in matplotlib.
If you're handling continuous values, you can try the snippet below to orient the colorbar.

```python
from matplotlib.colors import Normalize
from matplotlib import cm

import geopandas as gpd
gdf = gpd.read_file(gpd.datasets.get_path('naturalearth_lowres')) 

ax = gdf.plot(column='gdp_md_est', legend=False)
mn = gdf.gdp_md_est.min()
mx = gdf.gdp_md_est.max()
norm = Normalize(vmin=mn, vmax=mx)
n_cmap = cm.ScalarMappable(norm=norm, cmap="viridis")
n_cmap.set_array([])
ax.get_figure().colorbar(n_cmap, ax=ax, orientation='horizontal')
```
HTH

**Comment by mrahim** [comment]
BTW, I'm wondering whether adding `colorbar_kwds` in `plot_dataframe` would be useful ?

**Comment by jorisvandenbossche** [comment]
Or, we could pass the `legend_kwds` to the colorbar? (since it's either a legend or either a colorbar depnding on the data, and `legend=True` targets both)

**Comment by b-tierney** [comment]
I'm so sorry about the slow response – that snippet did the trick, of course, I was just being dumb.

In any case, colorbar_kwds would certainly be useful and intuitive for an already spectacular tool.

Thanks again, really appreciate the assist.

**Comment by jorisvandenbossche** [comment]
Let's leave it open for implementing the ability to pass keywords to the colorbar construction (by either passing the existing `legend_kwds` or by adding a `colorbar_kwds`)

**Comment by StevenLi-DS** [comment]
I have an issue that may relate to this. I'm trying to set the midplot of a color to `0`.
```python
import numpy as np
import matplotlib.pyplot as plt
import matplotlib.colors as colors
import geopandas as gpd

plt.style.use('seaborn-white')

# generate data
gdf = gpd.read_file(gpd.datasets.get_path('naturalearth_lowres'))
gdf = gdf[gdf.continent == 'Africa']
gdf['random'] = np.random.gamma(3, 2, len(gdf)) - 3

vmin, vmax, vcenter = gdf.random.min(), gdf.random.max(), 0
divnorm = colors.DivergingNorm(vmin=vmin, vcenter=vcenter, vmax=vmax)

fig, (ax1, ax2) = plt.subplots(1, 2, figsize=(15, 7))
map_1 = gdf.plot(column='random', cmap='RdBu', legend=True, ax=ax1)
map_2 = gdf.plot(column='random', cmap='RdBu', legend=True,
                 norm=divnorm, ax=ax2)

# fig.colorbar(map_2, ax=ax2)
```
![download](https://user-images.githubusercontent.com/33796896/61926338-2ed42900-af3e-11e9-91e6-46e9d940adce.png)

If I comment out the last line of code, I will get the following error.
```
fig, (ax1, ax2) = plt.subplots(1, 2, figsize=(15, 7))...
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
 in 
      8 
      9 
---> 10 ax2.get_figure().colorbar(map_2)

~\Anaconda3\lib\site-packages\matplotlib\figure.py in colorbar(self, mappable, cax, ax, use_gridspec, **kw)
   2213                              'panchor']
   2214         cb_kw = {k: v for k, v in kw.items() if k not in NON_COLORBAR_KEYS}
-> 2215         cb = cbar.colorbar_factory(cax, mappable, **cb_kw)
   2216 
   2217         self.sca(current_ax)

~\Anaconda3\lib\site-packages\matplotlib\colorbar.py in colorbar_factory(cax, mappable, **kwargs)
   1638         cb = ColorbarPatch(cax, mappable, **kwargs)
   1639     else:
-> 1640         cb = Colorbar(cax, mappable, **kwargs)
   1641 
   1642     cid = mappable.callbacksSM.connect('changed', cb.on_mappable_changed)

~\Anaconda3\lib\site-packages\matplotlib\colorbar.py in __init__(self, ax, mappable, **kw)
   1156         # Ensure the given mappable's norm has appropriate vmin and vmax set
   1157         # even if mappable.draw has not yet been called.
-> 1158         if mappable.get_array() is not None:
   1159             mappable.autoscale_None()
   1160 

AttributeError: 'AxesSubplot' object has no attribute 'get_array'

```
I've tried with other matplotlib plots, like `ax.scatter`, which works fine. Am I missing something?

**Comment by StevenLi-DS** [comment]
I think there should also be a `colorbar=Ture` instead of `legend=True` since the color doesn't really believe like a legend.

**Comment by StevenLi-DS** [comment]
I encounter this quite often that folks seek a way to set the midpoint of a colorbar to `0`. Please consider adding an arg in the `olorbar_kwds` like `dict(center=0)` or something to support this:

```python
import geopandas as gpd
import matplotlib.pyplot as plt
import matplotlib.colors as colors


# generate data
gdf = gpd.read_file(gpd.datasets.get_path('naturalearth_lowres'))
gdf['random'] = np.random.gamma(2, 2, len(gdf)) - 2

# normalize color
vmin, vmax, vcenter = gdf.random.min(), gdf.random.max(), 0
divnorm = colors.DivergingNorm(vmin=vmin, vcenter=vcenter, vmax=vmax)
# create a normalized colorbar
cbar = plt.cm.ScalarMappable(norm=divnorm, cmap='RdBu')

# plot
fig, ax = plt.subplots(figsize=(15, 7))
# with no normalization
gdf.plot(column='random', cmap='RdBu', legend=False, norm=divnorm, ax=ax)

# add colorbar
fig.colorbar(cbar, ax=ax)
```
![download](https://user-images.githubusercontent.com/33796896/61963146-0299c600-af99-11e9-9ec0-15ff4dac7168.png)

**Comment by martinfleis** [comment]
We can think about adding keywords for controlling this behaviour, feel free to draft a PR. 

Regarding your example, personally I would not add it to docs, but we can make an example out of it. However, not in a way you showed as color scheme is not the same in blues and reds (dark blue is much more extreme value than dark red), so the map is slightly misleading in this case.

**Comment by StevenLi-DS** [comment]
@martinfleis Could you label this issue as PR wanted and or something?

**Comment by martinfleis** [comment]
Keywords implemented in #1102. Thank you for raising this issue.

**Comment by StevenLi-DS** [comment]
Is there anyway to access the colorbar after the fact? For example:

```python
import geopandas as gpd
import matplotlib.pyplot as plt

gdf = gpd.read_file(gpd.datasets.get_path('naturalearth_lowres'))

fig, ax = plt.subplots(figsize=(10, 10))
cax = ax.inset_axes([0.63, 0.92, 0.03, 0.23], frameon=False)
gdf.plot(column='gdp_md_est', legend=True, cax=cax, ax=ax)
```
How do you get access to the colorbar geopandas plot created? I've tried ax.collections and it doesn't work.
